workflow/scripts/sleuth-init.R

```diff
@@ -37,18 +37,19 @@ if(!is.null(snakemake@params[["exclude"]])) {
 
 if(!is.null(model)) {
     # retrieve the model formula
-    formula <- as.formula(snakemake@params[["model"]])
+    formula <- as.formula(model)
     # extract variables from the formula and unnest any nested variables
     variables <- labels(terms(formula)) %>%
                     strsplit('[:*]') %>%
                     unlist()
     # select the columns required by sleuth and filter to all samples where
     # none of the given variables are NA
     samples <- samples %>%
-                select(sample, path, condition, variables) %>%
+                select(sample, path, all_of(variables)) %>%
```

Comment:
Can you please ensure that afterwards all columns that are not part of the formula are still in the sample sheet? We just want to remove rows here, not columns. The reason is that people might want to use certain columns e.g. in the bootstrap or PCA plots, even if they are not part of the formula.

Reply:
This is now in PR #19